src/csharp/Grpc.IntegrationTesting/ClientRunners.cs

```diff
@@ -55,6 +58,15 @@ public class ClientRunners
     {
         static readonly ILogger Logger = GrpcEnvironment.Logger.ForType<ClientRunners>();
 
+        // Profilers to use for clients.
+        static readonly BlockingCollection<BasicProfiler> profilers = new BlockingCollection<BasicProfiler>();
+
+        internal static void AddProfiler(BasicProfiler profiler)
```

Comment:
I am not super happy with this solution, but the idea is that you need one profiler per thread and in the program's Main, you need to dump profiler's data to a text file right before the application exits and you need access to the profiler from there.

So in main, you create the profiler, add it with ClientRunners.AddProfiler() and once everything's finished, you can do profiler.Dump("somefile.txt").